### PR TITLE
Walkthroughs implementation

### PR DESCRIPTION
**Pull request by JiglioNero**
#1688 
We have implemented the getting started guide for the extension using Walkthroughs API

image
image
image
image
image
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,6 +48,69 @@
...
+            "description": "Make sure you set up your environment correctly according to the guide described [here](https://reactnative.dev/docs/environment-setup)",
+            "media": {
+              "image": "media/environment-set-up.png",
+              "altText": "Empty image"
+            }
+          },
+          {
+            "id": "settingUpWindowsEnvironment",
+            "title": "Setting up the environment for Windows app development",
+            "description": "To develop React Native for Windows applications, you need to set up your environment in accordance with the requirements described [here](https://microsoft.github.io/react-native-windows/docs/rnw-dependencies)",
```
Comment (RedMickey) [inline package.json:69]:
```suggestion
            "description": "To develop React Native for Windows applications, you need to set up your dev environment in accordance with the [official requirements](https://microsoft.github.io/react-native-windows/docs/rnw-dependencies)",
```

File: package.json [hunk 2]
```diff
@@ -48,6 +48,69 @@
...
+            "when": "isWindows",
+            "media": {
+              "image": "media/rnw-set-up.png",
+              "altText": "Empty image"
+            }
+          },
+          {
+            "id": "settingUpMacOsEnvironment",
+            "title": "Setting up the environment for macOS app development",
+            "description": "To develop React Native for macOS applications, you need to set up your environment in accordance with the requirements described [here](https://microsoft.github.io/react-native-windows/docs/rnm-dependencies)",
```
Comment (RedMickey) [inline package.json:79]:
```suggestion
            "description": "To develop React Native for macOS applications, you need to set up your dev environment in accordance with the [official requirements](https://microsoft.github.io/react-native-windows/docs/rnm-dependencies)",
```

File: media/features-tutor.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Learn main features
+## Commands of the Command Palette
```
Comment (RedMickey) [inline media/features-tutor.md:2]:
```suggestion
## React Native commands in the Command Palette
```

File: media/features-tutor.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+This extension supports many `commands of the Command Palette` for running and debugging React Native applications, and other supporting commands for convenient use of the extension. The complete list of the command you can see in the corresponding [paragraph](https://github.com/microsoft/vscode-react-native#react-native-commands-in-the-command-palette) of the documentation.
```
Comment (RedMickey) [inline media/features-tutor.md:3]:
```suggestion
The extension provides special `Command Palette commands` for running React Native applications on different platforms, operating with the Packager and automating some dev processes. You can find more info about supported commands [here](https://github.com/microsoft/vscode-react-native#react-native-commands-in-the-command-palette).
```

File: media/features-tutor.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+## Network inspector
+The extension provides `Network inspector` feature to inspect outgoing network traffic in your apps. You can browse all requests being made and their responses in VS Code DevTools console. Environment requirements and other details for using the Network inspector can be found in [this](https://github.com/microsoft/vscode-react-native#network-inspector) paragraph of the documentation.
```
Comment (RedMickey) [inline media/features-tutor.md:5]:
```suggestion
The extension provides `Network inspector` feature to inspect outgoing network traffic in your apps. You can browse all requests being made and their responses in VS Code DevTools console. More details about Network inspector usage and its requirements are available [here] (https://github.com/microsoft/vscode-react-native#network-inspector) in the documentation.
```

File: media/features-tutor.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+## Element inspector
+With the command `reactNative.command.runInspector` you can launch React DevTools to conveniently work with the DOM-tree of your application. Details on how you can install and use React DevTools in this extension can be found [here](https://github.com/microsoft/vscode-react-native#element-inspector).
```
Comment (RedMickey) [inline media/features-tutor.md:7]:
```suggestion
In the extension you can run React DevTools Element inspector to inspect the DOM-tree of your application by starting `Run Element Inspector` command in Command Palette. You can learn more about React DevTools installation and usage [here](https://github.com/microsoft/vscode-react-native#element-inspector).
```

File: README.md [hunk 7]
```diff
@@ -773,13 +773,27 @@ The extension provides context variables for the following features:
 |`isRNTNetworkInspectorRunning` - true if the Network inspector is running|Network inspector|
 
 Using these context variables you can assign the same keyboard combination for some pair commands for more convenient use. For example, you can configure the same key bindings for `Start Packager` and `Stop Packager` commands using `when` clauses, as shown below:
-![image](images/custom-keybindings.png)
+![image](resources/images/custom-keybindings.png)
+
+# Element inspector
+
+With the help of the [`react-devtools`](https://www.npmjs.com/package/react-devtools) package and this extension, you can conveniently explore and work with the DOM-tree.
```
Comment (RedMickey) [inline README.md:780]:
```suggestion
In the extension you can run React DevTools Element inspector to inspect the DOM-tree of your application.
It's required to install the [`react-devtools`](https://www.npmjs.com/package/react-devtools) package globally to run React DevTools from the extension.
\```
npm install -g react-devtools
\```
```

File: README.md [hunk 8]
```diff
@@ -773,13 +773,27 @@ The extension provides context variables for the following features:
...
+
+### Integration with React Native Inspector
+
+You can open the in-app developer menu and choose "Show Inspector". It will bring up an overlay that lets you tap on any UI element and see information about it:
+
+![Element inspector usage](media\element-inspector-usage.gif)
+
+Using the command of Command Palette `reactNative.command.runInspector` you can launch DevTools. Inspector will enter a special collapsed mode, and instead use the DevTools as primary UI. In this mode, clicking on something in the simulator will bring up the relevant components in the DevTools:
```
Comment (RedMickey) [inline README.md:788]:
```suggestion
Using the `Run Element Inspector` Command Palette command you can launch DevTools in the extension. Inspector will enter a special collapsed mode, and use the DevTools as primary UI. In this mode, clicking on something in the simulator will bring up the relevant components in the DevTools:
```

File: package.json [hunk 9]
```diff
@@ -48,6 +48,76 @@
   ],
   "main": "./src/extension/rn-extension",
   "contributes": {
+    "walkthroughs": [
+      {
+        "id": "RNTGetStarted",
+        "title": "Get started with React Native development",
+        "description": "Set up the environment for React Native development and explore extension functionality",
```
Comment (RedMickey) [inline package.json:55]:
```suggestion
        "description": "Your first steps to set up the environment for React Native development and explore 
the extension functionality",
```

File: package.json [hunk 10]
```diff
@@ -48,6 +48,76 @@
...
+        "steps": [
+          {
+            "id": "settingUpCommonEnvironment",
+            "title": "Set up the development environment",
```
Comment (RedMickey) [inline package.json:59]:
```suggestion
            "title": "Set up RN development environment",
```

File: package.json [hunk 11]
```diff
@@ -48,6 +48,76 @@
...
+            "description": "For the React Native Tools extension to work correctly, you need to set up the development environment in accordance with the requirements of React Native\n[Learn more](https://reactnative.dev/docs/environment-setup)",
```
Comment (RedMickey) [inline package.json:60]:
```suggestion
            "description": "To develop React Native applications in the React Native Tools extension, you need to set up your development environment in accordance with the requirements of React Native\n[Learn more](https://reactnative.dev/docs/environment-setup)",
```

File: package.json [hunk 12]
```diff
@@ -48,6 +48,76 @@
...
+            "media": {
+              "markdown": "resources/walkthougths/setup-env-common.md"
+            }
+          },
+          {
+            "id": "settingUpExpoEnvironment",
+            "title": "Set up the development environment",
```
Comment (RedMickey) [inline package.json:67]:
```suggestion
            "title": "Prepare for Expo development",
```

File: package.json [hunk 13]
```diff
@@ -48,6 +48,76 @@
...
+            "description": "The extension allows you to conveniently work with Expo applications. To get started, you need to set up the development environment in accordance with the official requirements\n[Learn more](https://reactnative.dev/docs/environment-setup)",
```
Comment (RedMickey) [inline package.json:68]:
```suggestion
            "description": "The extension allows you to conveniently work with Expo applications. To get started, you need to set up the development environment in accordance with the extension requirements\n[Learn more](https://reactnative.dev/docs/environment-setup)",
```

File: package.json [hunk 14]
```diff
@@ -48,6 +48,76 @@
...
+            "description": "The Hermes engine is an open source JavaScript engine created by Facebook to optimize building and running React Native applications. It improves app performance and decreases app size.\n[Learn more](https://github.com/microsoft/vscode-react-native#hermes-engine)",
+            "media": {
+              "image": "resources/images/hermeslogo.svg",
+              "altText": "Empty image"
+            }
+          },
+          {
+            "id": "runAndDebugReactNativeApp",
+            "title": "Run and Debug your React Native application",
+            "description": "Create a debug configuration and start debugging your React Native applications.\n[Learn more](https://github.com/microsoft/vscode-react-native#debugging-react-native-applications)",
```
Comment (RedMickey) [inline package.json:105]:
```suggestion
            "description": "Create a debug configuration and start debugging your React Native application.\n[Learn more](https://github.com/microsoft/vscode-react-native#debugging-react-native-applications)",
```

File: package.json [hunk 15]
```diff
@@ -48,6 +48,76 @@
...
+            "media": {
+              "markdown": "resources/walkthougths/debug-tutor.md"
+            }
+          },
+          {
+            "id": "learnMainFeatures",
+            "title": "Learn main features",
+            "description": "The extension supports all the tools you need to quickly and easily develop your applications",
```
Comment (RedMickey) [inline package.json:113]:
```suggestion
            "description": "The extension provides useful features to facilitate the development of React Native applications",
```

File: resources/walkthougths/debug-tutor.md [hunk 16]
```diff
@@ -0,0 +1,21 @@
+# Run and Debug your React Native application
+## Ways to start debug session
```
Comment (RedMickey) [inline resources/walkthougths/debug-tutor.md:2]:
```suggestion
## Create and run debugging scenario
```

File: resources/walkthougths/debug-tutor.md [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+This extension supports generating debugging configuration in the `.vscode/launch.json` file by selecting debugging parameters in dropdown lists at the top of the editor.
+
+![Add React Native debug configuration](../images/add-debug-configuration.gif)
+
```
Comment (RedMickey) [inline resources/walkthougths/debug-tutor.md:6]:
```suggestion

To start debugging, just choose the appropriate configuration in the Configuration dropdown, and then press F5 (or click _Green Arrow_ ![Configure-gear](resources/images/debug-icon.png) button) to start debugging your app in VS Code.

```

File: resources/walkthougths/setup-env-common.md [hunk 18]
```diff
@@ -0,0 +1,10 @@
+# Getting started
+Before going any further, please review the following information to help you successfully work with React Native Tools.
```
Comment (RedMickey) [inline resources/walkthougths/setup-env-common.md:2]:
```suggestion
Before going any further, please check out the following information to help you get started with React Native Tools successfully.
```

File: resources/walkthougths/setup-env-common.md [hunk 19]
```diff
@@ -0,0 +1,10 @@
...
+
+Make sure that you:
+
+- [have a working React Native environment](https://reactnative.dev/docs/environment-setup).
+- have the [`emulator`](https://developer.android.com/studio/run/emulator-commandline) utility available in your `PATH` if you're developing Android applications using emulators
```
Comment (RedMickey) [inline resources/walkthougths/setup-env-common.md:7]:
```suggestion
- have the [`emulator`](https://developer.android.com/studio/run/emulator-commandline) utility available in your `PATH` if you're developing Android applications using emulators.
```

File: package.json [hunk 20]
```diff
@@ -48,6 +48,76 @@
...
+            "title": "Set up RN development environment",
+            "description": "To develop React Native applications in the React Native Tools extension, you need to set up your development environment in accordance with the requirements of React Native\n[Learn more](https://reactnative.dev/docs/environment-setup)",
+            "media": {
+              "markdown": "resources/walkthougths/setup-env-common.md"
+            }
+          },
+          {
+            "id": "settingUpExpoEnvironment",
+            "title": "Prepare for Expo development",
+            "description": "The extension allows you to conveniently work with Expo applications. To get started, you need to set up the development environment in accordance with the extension requirements\n[Learn more](https://github.com/microsoft/vscode-react-native#expo-applications)",
```
Comment (RedMickey) [inline package.json:68]:
```suggestion
            "description": "The extension provides functionality for developing Expo applications. To get started, you need to set up the development environment in accordance with the extension requirements\n[Learn more](https://github.com/microsoft/vscode-react-native#expo-applications)",
```

File: resources/walkthougths/setup-env-expo.md [hunk 21]
```diff
@@ -0,0 +1,9 @@
+# Expo applications
+Prepare your environment by following the [Expo CLI Quickstart instruction](https://reactnative.dev/docs/environment-setup).
+Also pay attention to the `React Native CLI Quickstart` tab, where you can find other requirements. Also you need to:
+- Install the [Expo app](https://getexponent.com/) on the target device or emulator
```
Comment (RedMickey) [inline resources/walkthougths/setup-env-expo.md:4]:
```suggestion
- Install the [Expo app](https://docs.expo.dev/get-started/installation/#2-expo-go-app-for-ios-and) on the target device or emulator
```

File: resources/walkthougths/debug-tutor.md [hunk 22]
```diff
@@ -0,0 +1,23 @@
...
+The extension also allows to start debugging without creating the `launch.json` file in one of the following ways:
+- Using dynamic debugging configurations
+
+    ![Run dynamic debugging configuration](../images/dynamic-debugging-configuration.gif)
+- Using Debug button in the Editor Title area menu
+
+    ![Select and run debugging command](../images/debug-commands-button.png)
+
+## Enable `Debug JS Remotely` and attach to application
```
Comment (RedMickey) [inline resources/walkthougths/debug-tutor.md:17]:
```suggestion
## Enable debugging mode and attach to the application
```

File: package.json [hunk 23]
```diff
@@ -48,6 +48,76 @@
...
+            "id": "runAndDebugReactNativeApp",
+            "title": "Run and Debug your React Native application",
+            "description": "Create a debug configuration and start debugging your React Native application.\n[Learn more](https://github.com/microsoft/vscode-react-native#debugging-react-native-applications)",
+            "media": {
+              "markdown": "resources/walkthougths/debug-tutor.md"
+            }
+          },
+          {
+            "id": "learnMainFeatures",
+            "title": "Learn main features",
```
Comment (RedMickey) [inline package.json:112]:
```suggestion
            "title": "Explore expansion features",
```

File: resources/walkthougths/features-tutor.md [hunk 24]
```diff
@@ -0,0 +1,19 @@
...
+The extension provides special `Command Palette commands` for running React Native applications on different platforms, operating with the Packager and automating some dev processes. You can find more info about supported commands [here](https://github.com/microsoft/vscode-react-native#react-native-commands-in-the-command-palette).
+## Network inspector
+The extension provides `Network inspector` feature to inspect outgoing network traffic in your apps. You can browse all requests being made and their responses in VS Code DevTools console. More details about Network inspector usage and its requirements are available [here](https://github.com/microsoft/vscode-react-native#network-inspector) in the documentation.
+## Element inspector
+In the extension you can run React DevTools Element inspector to inspect the DOM-tree of your application by starting `Run Element Inspector` command in Command Palette. You can learn more about React DevTools installation and usage [here](https://github.com/microsoft/vscode-react-native#element-inspector).
+## Customization
+Also, the extension can be customized for more convenient work with your applications. Examples of customizable features:
```
Comment (RedMickey) [inline resources/walkthougths/features-tutor.md:9]:
```suggestion
The extension provides different settings that can be customized according to personal preferences to improve the extension usability. Please see the [Customization](https://github.com/microsoft/vscode-react-native#customization) section in the documentation.
```

File: resources/walkthougths/setup-env-expo.md [hunk 25]
```diff
@@ -0,0 +1,10 @@
+# Expo applications
+Prepare your environment by following the [Expo CLI Quickstart instruction](https://reactnative.dev/docs/environment-setup).
+For correct work with Expo this extension **`requires Android SDK`**.
+So also pay attention to the `React Native CLI Quickstart` tab, where you can find the Android SDK installation guide:
+
+- Install the [Expo app](https://getexponent.com/) on the target device or emulator
```
Comment (RedMickey) [inline resources/walkthougths/setup-env-expo.md:6]:
```suggestion
- Install the [Expo app](https://docs.expo.dev/get-started/installation/#2-expo-go-app-for-ios-and) on the target device or emulator
```

File: resources/walkthougths/features-tutor.md [hunk 26]
```diff
@@ -0,0 +1,9 @@
+# Learn main features
```
Comment (RedMickey) [inline resources/walkthougths/features-tutor.md:1]:
```suggestion
# Main extension features
```

File: package.json [hunk 27]
```diff
@@ -48,6 +48,76 @@
...
+            }
+          },
+          {
+            "id": "settingUpWindowsEnvironment",
+            "title": "Prepare for React Native for Windows",
+            "description": "To develop React Native for Windows applications, you need to set up your dev environment in accordance with the [official requirements](https://microsoft.github.io/react-native-windows/docs/rnw-dependencies)",
+            "when": "isWindows",
+            "media": {
+              "image": "resources/images/rnw-set-up.png",
+              "altText": "React Native for Windows setup"
```
Comment (RedMickey) [inline package.json:80]:
```suggestion
              "altText": "React Native for Windows app view"
```

File: package.json [hunk 28]
```diff
@@ -48,6 +48,76 @@
...
+            }
+          },
+          {
+            "id": "settingUpMacOsEnvironment",
+            "title": "Prepare for React Native for macOS",
+            "description": "To develop React Native for macOS applications, you need to set up your dev environment in accordance with the [official requirements](https://microsoft.github.io/react-native-windows/docs/rnm-dependencies)",
+            "when": "isMac",
+            "media": {
+              "image": "resources/images/rnm-set-up.png",
+              "altText": "React Native for macOS setup"
```
Comment (RedMickey) [inline package.json:90]:
```suggestion
              "altText": "React Native for macOS app view"
```

**Comment by JiglioNero** [comment]
Hi @molant @EricCornelson Could you please take a look at this?

**Review by RedMickey — APPROVED** [review]

**Pull request merged**